driver/sum_driver.hpp

```diff
@@ -290,19 +340,14 @@ int SumDriver<Tgpu, Tref>::RunBackwardGPU()
 template <typename Tgpu, typename Tref>
 Tref SumDriver<Tgpu, Tref>::GetTolerance()
 {
-    if(data_type == miopenHalf)
-    {
-        return 1e-3;
-    }
-    else if(data_type == miopenFloat)
-    {
-        return 5e-5;
-    }
-    else if(data_type == miopenBFloat16)
-    {
-        return 1e-2;
-    }
-    return 0;
+    // Computation error of fp16 is ~2^13 (=8192) bigger than
+    // the one of fp32 because mantissa is shorter by 13 bits.
+    auto tolerance = (sizeof(Tgpu) == 4 || sizeof(Tgpu) == 1) ? 1.5e-6 : 8.2e-3;
+
+    // bf16 mantissa has 7 bits, by 3 bits shorter than fp16.
+    if(std::is_same<Tgpu, bfloat16>::value)
+        tolerance *= 8.0;
```

Comment:
@seungmanhan This is correct. Sorry for errors in https://github.com/ROCm/MIOpen/pull/2543#discussion_r1402038388 (fixed now)

[R] `sizeof` here is outdated technique (and it may match `INT32` type), please prefer `is_same` next time.

[Q] I see `sizeof(Tgpu) == 1` -- do you mean F8 or INT8 type here?

Reply:
In this PR I don't use F8 or INT8. I will reflect this correction in Argmax PR. https://github.com/ROCm/MIOpen/pull/2583